ucsdscheduleme/Repo/FormatRepo.cs

```diff
@@ -0,0 +1,222 @@
+
+namespace ucsdscheduleme.Repo
+{
+    public class FormatRepo
+    {
+        private static readonly int START_HOUR = 7;
+        private static readonly int START_MINUTES = 30;
+        private static readonly int MINUTES_IN_HOUR = 60;
+
+        public static ScheduleViewModel FormatSectionsToCalendarEvent(List<Section> selectedSections)
+        {
+            ScheduleViewModel model = new ScheduleViewModel
+            {
+                Courses = new Dictionary<int, CourseViewModel>()
+            };
+            var coursesSectionPairs = selectedSections.ToDictionary(s => s,s => s.Course);
+
+            foreach (var courseSection in coursesSectionPairs)
+            {
+                var course = courseSection.Value;
+                var selectedSection = courseSection.Key;
+                CourseViewModel thisCourse = new CourseViewModel
+                {
+                    Bases = new Dictionary<char, BaseViewModel>(),
+                    SelectedSection = selectedSection.Id,
+                    // If our sections have no meetings, we're in trouble. Time to abort.
+                    SelectedBase = selectedSection.Meetings?.First().Code[0] ?? 
+                        throw new ArgumentException($"Section with no meeting found: Id'{selectedSection.Id}'")
+                };
+
+                model.Courses.Add(course.Id,thisCourse);
+
+                var baseSectionGroups = course.Sections.GroupBy(s => s.Meetings.First().Code[0]);
+                foreach (var baseSectionGroup in baseSectionGroups)
+                {
+                    var baseForCourse = PopulateBaseForCourse(selectedSection, baseSectionGroup.ToList());
+                    thisCourse.Bases.Add(baseSectionGroup.Key, baseForCourse);
+                }
+            }
+
+            return model;
+        }
+
+        private static BaseViewModel PopulateBaseForCourse(Section selectedSection, List<Section> sectionsForBase)
+        {
+            var course = selectedSection.Course;
+
+            BaseViewModel thisBase = new BaseViewModel
+            {
+                Metadata = AddMetadata(sectionsForBase.First())
+            };
+
+            List<OneTimeEvent> oneTimeEvents = PopulateOneTimeEvents(sectionsForBase, course);
+            thisBase.OneTimeEvents = oneTimeEvents;
+
+            List<CalendarEvent> baseEvents = PopulateBaseEvents(selectedSection, sectionsForBase, course);
+            thisBase.BaseEvents = baseEvents;
+
+            Dictionary<int, List<CalendarEvent>> thisBasesSectionEvents = PopulateSectionEventsForBase(sectionsForBase, course);
+            thisBase.SectionEvents = thisBasesSectionEvents;
+
+            return thisBase;
+        }
+
+        private static Dictionary<int, List<CalendarEvent>> PopulateSectionEventsForBase(List<Section> sectionsForBase, Course course)
+        {
+            Dictionary<int, List<CalendarEvent>> thisBasesSectionEvents = new Dictionary<int, List<CalendarEvent>>();
```

Comment:
explaining what the key value is would be helpful even if you can tell from the following code like line 80